src/dmd/dtemplate.d

```diff
@@ -5697,6 +5697,10 @@ extern (C++) final class TemplateAliasParameter : TemplateParameter
                  *  template X(T) {}        // T => sa
                  */
             }
+            else if (global.params.allowTypesForTemplateAlias && sa && cast(Type) sa)
```

Comment:
`sa` is an `extern(C++)` class, hence the third condition (`cast(Type) sa`) does not do a dynamic cast, but a reinterpret one.

Reply:
Oh f*ck.

Reply:
Better?